### PR TITLE
Add blog post about Trino innovation

### PR DESCRIPTION
**Pull request by mosabua**

### CODE REVIEW

**Comment by mosabua** [comment]
General question.. should we link to focussed guides where we have them rather than the website of a client too .. e.g. https://docs.starburst.io/data-consumer/clients/superset.html , https://docs.starburst.io/data-consumer/clients/dbeaver.html and https://docs.starburst.io/data-consumer/clients/metabase.html and others .. or even add as separate link

File: _posts/2022-08-04-decade-innovation.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+efficiency and low latency were a core design goal, and with features like
+native compilation the delivered performance superseded other systems. Over the
+years our query analyzer and planner was supplemented by more and more
+sophisticated algorithms and features. Connectors learned to retrieve and manage
+table statistics, the optimizer was created and morphed into a [cost-based
+optimizer]({% post_url 2019-07-04-cbo-introduction %}), and we added further
+improvements that benefit query processing performance. We added dynamic
+filtering, [dynamic partition pruning]({% post_url
+2020-06-14-dynamic-partition-pruning %}), predicate pushdown, join pushdown,
+aggegrate function pushdown and numerous others. Each of these improvements was
```
Comment (wendigo) [inline _posts/2022-08-04-decade-innovation.md:158]: Aggregate

Comment (mosabua) [reply _posts/2022-08-04-decade-innovation.md:158]: oops.. typo fixed

File: _posts/2022-08-04-decade-innovation.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+starting with LDAP and password file, to the modern OAuth2.0 and SSO systems.
+Trino supports fine-grained access control and [security management SQL commands
+like `GRANT` and
+`REVOKE`](https://trino.io/docs/current/language/sql-support.html#security-operations).
+You can secure connections from client tools, and use numerous methods to ensure
+secured access to your data sources.
+
+## Client tools and integrations
+
+In the very beginning all you could do it submit a query to the [client REST
```
[the remark this answers is not in the document]
Comment (mosabua) [reply _posts/2022-08-04-decade-innovation.md:62]: yep

File: _posts/2022-08-04-decade-innovation.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
+---
+layout: post
+title:  "A decade of query engine innovation"
+author: "Manfred Moser, Martin Traverso, Dain Sundstrom, David Phillips"
+excerpt_separator: <!--more-->
+---
+
+Amazing how far we have come! Our massively-parallel processing SQL query
```
Comment (bitsondatadev) [inline _posts/2022-08-04-decade-innovation.md:8]:
```suggestion
It's amazing how far we have come! Our massively-parallel processing SQL query
```

File: _posts/2022-08-04-decade-innovation.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+Starting with how you even run Trino and install it, numerous changes came about
+in the last decade. We moved from Java 7 to Java 8, then to Java 11, and [only
+recently to the latest supported Java LTS release - Java 17]({% post_url
+2022-07-14-trino-updates-to-java-17 %}). Each time we
+benefited from the innovations in the runtime performance as well as the
+improved Java language features. With **Java 17**, we are just about to start a lot
+of these improvements.
+
+When it comes to actually running and deploying Trino, the **tarball** is still a
```
Comment (bitsondatadev) [inline _posts/2022-08-04-decade-innovation.md:28]:
```suggestion
When it comes to actually [running and deploying Trino]({{site.url}}/episodes/35.html), the **tarball** is still a
```

File: _posts/2022-08-04-decade-innovation.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+advantage of our added support for the **ARM processor architecture** and the
+increasing availability of suitable servers from the cloud providers.
+
+## Security
+
+What is security, authentication, authorization? In the beginning none of this
+existed in the first releases of Trino. Two years after launch we added first
+simple authentication and authorization support. Today the days when Kerberos
+was critical, and you needed to use the Java KeyStore in most deployments are
+long gone. The wide adoption of Trino led to improvements such as support for
```
Comment (bitsondatadev) [inline _posts/2022-08-04-decade-innovation.md:46]:
```suggestion
simple authentication and authorization support. Long gone are the days when
Kerberos was critical, and you needed to use the Java KeyStore in most 
deployments. The wide adoption of Trino led to improvements such as support for
```

File: _posts/2022-08-04-decade-innovation.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+That’s all done and `MERGE` is next. But the core language features were not
+able to satisfy the needs of our users. We added functions for a large variety
+of topics ranging from simple string and [date
+functions](https://trino.io/docs/current/functions/datetime.html) to [JSON
+support](https://trino.io/docs/current/functions/json.html), [geospatial
+functions](https://trino.io/docs/current/functions/geospatial.html), and many
+others.
+
+From the core language perspective we added newer SQL functionality, such as
+[window functions and row pattern matching support]({% post_url
```
Comment (bitsondatadev) [inline _posts/2022-08-04-decade-innovation.md:99]:
```suggestion
[window functions and `MATCH_RECOGNIZE` support]({% post_url
```

Comment (bitsondatadev) [reply _posts/2022-08-04-decade-innovation.md:99]: I've had people specifically call out this function as a reason they came to Trino. I think we should mention the SQL keyword. WDYT?

File: _posts/2022-08-04-decade-innovation.md [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+
+One large pivot we recently added was the addition of [fault-tolerant query
+execution mode]({% post_url 2022-05-05-tardigrade-launch %}). Queries no longer
+die from a minor failure with this feature enabled. Parts of the execution can
+be retried and query processing can proceed. Trino is moving on from the best
+analytics engine to be the best query engine overall!
+
+## Looking forward
+
+As you can see there is a lot to look back to, and celebrate. But while we are
```
Comment (bitsondatadev) [inline _posts/2022-08-04-decade-innovation.md:170]:
```suggestion
As you can see there is a lot to look back to and celebrate. But while we are
```

Comment (bitsondatadev) [reply _posts/2022-08-04-decade-innovation.md:170]: "and celebrate" is not a separate clause. The comma seems strange here.

**Review by bitsondatadev — COMMENTED** [review]
Looks good..few small nits and additions

File: _posts/2022-08-04-decade-innovation.md (hunk 5 above)
Comment (mosabua) [reply _posts/2022-08-04-decade-innovation.md:46]: I had that before but it doesnt work as well with the prior sentence

**Review by bitsondatadev — APPROVED** [review]
LGTM!

File: _posts/2022-08-04-decade-innovation.md [hunk 8]
```diff
@@ -0,0 +1,182 @@
+---
+layout: post
+title:  "A decade of query engine innovation"
+author: "Manfred Moser, Martin Traverso, Dain Sundstrom, David Phillips"
+excerpt_separator: <!--more-->
+---
+
+It's amazing how far we have come! Our massively-parallel processing SQL query
+engine, Trino, has really grown up. We move beyond just querying object stores
```
Comment (martint) [inline _posts/2022-08-04-decade-innovation.md:9]: Should this be:

> We've moved

File: _posts/2022-08-04-decade-innovation.md [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+of these improvements.
+
+When it comes to actually [running and deploying
+Trino]({{site.url}}/episodes/35.html), the **tarball** is still a good choice
+for simple installation and as a base for other packages. Over time we added
+**RPM** archive support, which is being replaced more and more by Docker
+**containers**. The container images also enable modern deployment on Kubernetes
+with [our Helm chart](https://github.com/trinodb/charts).
+
+And let us add one last note about deployments. It’s been amazing to see how the
```
Comment (martint) [inline _posts/2022-08-04-decade-innovation.md:36]: Not sure what point this is trying to make:

> It’s been amazing to see how the machine sizes in terms of CPU and memory keep growing. Some truly large machines are now running as clusters.

If we're trying to say that Trino supports larger machines, then we should say so more directly. However, we haven't done anything specific for this purpose, so I might just remove it.

File: _posts/2022-08-04-decade-innovation.md [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+was critical, and you needed to use the Java KeyStore in most deployments are
+long gone. The wide adoption of Trino led to improvements such as support for
+[automatic certificate creation and TLS for internal
+communication](https://trino.io/docs/current/security/internal-communication.html),
+[secret injection from environment
+variables](https://trino.io/docs/current/security/secrets.html), and the many
+[authentication
+types](https://trino.io/docs/current/security/authentication-types.html)
+starting with LDAP and password file, to the modern OAuth2.0 and SSO systems.
+Trino supports fine-grained access control and [security management SQL commands
```
Comment (martint) [inline _posts/2022-08-04-decade-innovation.md:56]: Maybe also mention column masking and row-level security.

File: _posts/2022-08-04-decade-innovation.md [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+
+Looking at the number of available connectors, innovation has been tremendous.
+The original Hive connector with support for HDFS and a Hive Metastore Service,
+became a powerhouse of features. Support for object storage systems including
+Amazon S3 and compatible systems, Azure Data Lake Storage, and Google Cloud
+Storage, was supplemented by support for Amazon Glue as metastore. We also
+constantly added support for different file formats in these systems, and
+improved performance for ORC, Parquet, Avro, and others.
+
+The initial idea to support other data sources, led to connectors for over a
```
Comment (martint) [inline _posts/2022-08-04-decade-innovation.md:124]: Remove comma

File: _posts/2022-08-04-decade-innovation.md [hunk 12]
```diff
@@ -0,0 +1,182 @@
...
+dozen other databases including relational systems such
```
Comment (martint) [inline _posts/2022-08-04-decade-innovation.md:125]: Comma between "databases" and "including"

File: _posts/2022-08-04-decade-innovation.md [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+[Apache Kafka](https://kafka.apache.org/), and other systems that traditionally
+are not available to query with SQL. Trino unlocks completely new use cases for
+these systems.
+
+The wide range of supported systems includes traditional data lakes and data
+warehouses. With the emerging new table formats and the related Trino
+connectors, our project is a powerful tool to run your lakehouse system. [Delta
+Lake](https://delta.io/) and [Apache Iceberg](https://iceberg.apache.org/)
+connectors are already capable of full read and write operations and include
+numerous other features. A [Apache Hudi](https://hudi.apache.org/) connector is
```
Comment (martint) [inline _posts/2022-08-04-decade-innovation.md:140]: > A*n* Apache Hudi ...

File: _posts/2022-08-04-decade-innovation.md [hunk 14]
```diff
@@ -0,0 +1,182 @@
...
+We also have robust and widely used connectors for real-time analytics systems
+like [Apache Pinot](https://pinot.apache.org/), [Apache
+Druid](https://druid.apache.org/) and [Clickhouse](https://clickhouse.com/),
+that are constantly improved by the community.
+
+## Query processing and performance
+
+Last but not least, these queries also need to be processed. From the start high
+efficiency and low latency were a core design goal, and with features like
+native compilation the delivered performance superseded other systems. Over the
```
Comment (martint) [inline _posts/2022-08-04-decade-innovation.md:152]: Maybe

> the *resulting* performance

Comment (martint) [reply _posts/2022-08-04-decade-innovation.md:152]: s/superseded/surpassed

File: _posts/2022-08-04-decade-innovation.md [hunk 15]
```diff
@@ -0,0 +1,182 @@
...
+optimizer]({% post_url 2019-07-04-cbo-introduction %}), and we added further
+improvements that benefit query processing performance. We added dynamic
+filtering, [dynamic partition pruning]({% post_url
+2020-06-14-dynamic-partition-pruning %}), predicate pushdown, join pushdown,
+aggregate function pushdown and numerous others. Each of these improvements was
+also finely tuned, and runs in production with huge workloads providing us more
+data on how to improve next.
+
+One large pivot we recently added was the addition of [fault-tolerant query
+execution mode]({% post_url 2022-05-05-tardigrade-launch %}). Queries no longer
```
Comment (martint) [inline _posts/2022-08-04-decade-innovation.md:165]: > Queries execution can survive cluster node failures when this feature is enabled.

File: _posts/2022-08-04-decade-innovation.md [hunk 16]
```diff
@@ -0,0 +1,182 @@
...
+die from a minor failure with this feature enabled. Parts of the execution can
+be retried and query processing can proceed. Trino is moving on from the best
+analytics engine to be the best query engine overall!
```
Comment (martint) [inline _posts/2022-08-04-decade-innovation.md:168]: > the best query engine overall!

This is claiming too much -- Trino is not trying to become, say, an OLTP query engine

File: _posts/2022-08-04-decade-innovation.md [hunk 17]
```diff
@@ -0,0 +1,182 @@
...
+
+## Looking forward
+
+As you can see there is a lot to look back to and celebrate. But while we are
+definitely proud of our successes working with you all, we see no time to rest.
```
Comment (martint) [inline _posts/2022-08-04-decade-innovation.md:173]: Who is "you all" in this sentence?

File: _posts/2022-08-04-decade-innovation.md (hunk 9 above)
Comment (mosabua) [reply _posts/2022-08-04-decade-innovation.md:36]: Will reword

File: _posts/2022-08-04-decade-innovation.md (hunk 10 above)
Comment (mosabua) [reply _posts/2022-08-04-decade-innovation.md:56]: thats not in trino really

Comment (martint) [reply _posts/2022-08-04-decade-innovation.md:59]: It is. The engine has built in support for column masks and row filters, and a set of SPIs that access control implementations can build on to provide those masks and filters.

**Pull request merged**